### DISCUSSION TITLE
❓ Ask for help

### DISCUSSION THREAD

**Discussion by EngiBenchi** [louislam/dockge#226] [Ask for Help] [answered]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

While trying to start the container using sudo docker-compose up -d I'll get the error "mkdir /opt/stacks: read-only file system"
I'm the owner of the directory and set to 744
Has any one a suggestion what I am doing wrong?

### 📝 Error Message(s) or Log

ERROR: for 24a60cedff13_dockge_dockge_1  Cannot start service dockge: error while creating mount source path '/opt/stacks/dockge/data': mkdir /opt/stacks: read-only file system

### 🐻 Dockge Version

1

### 💻 Operating System and Arch

Ubuntu 22.04.3 LTS

### 🌐 Browser

Firefox

### 🐋 Docker Version

Docker version 20.10.24, build 297e128

### 🟩 NodeJS Version

v12.22.9

**Comment by louislam** [comment, marked as answer]
Not quite sure.

You can try to use another location like /stacks

You can download the compose.yaml here:
https://dockge.kuma.pet/compose.yaml?port=5001&stacksPath=/stacks

**Reply by EngiBenchi** [reply]
Thanks that did the trick. 
But I initially wanted to mount at /var/stacks because/var is mounted at its own ssd so more free space there. 
Any suggestions to get that working?

**Reply by louislam** [reply]
Not so familiar with permission issues.

But you can freely change the path like: `/var/stacks` to test it.
https://dockge.kuma.pet/compose.yaml?port=5001&stacksPath=/var/stacks

Or `/var/my_stacks`
https://dockge.kuma.pet/compose.yaml?port=5001&stacksPath=/var/my_stacks

**Reply by EngiBenchi** [reply]
Yes that is what I started doing. But it resulted in the same error.

**Reply by trythatagain** [reply]
Have tried 774 for the folder? I'm wondering whether the group needs access as well. I'm not to brilliant with permissions but I know I've had to make sure group is set as well. It's only a suggestion.

**Reply by EngiBenchi** [reply]
Thanks I will give that a try. 
I think it must be something like that I will compare the current folder settings with the one in /var to learn from it. Haven't had time to do so yet.

**Reply by EngiBenchi** [reply]
Okay I took a bit more time to check the folder /opt/stacks is set as own by me the login user en has 40755 as permission octal
And that is exactly the same as the map where it is working /stacks So I don't think it is with  something to do with my permission settings.

![image](https://github.com/louislam/dockge/assets/20003780/80d1f2dc-7837-4b06-af66-05ba7a3e515d)
I tried it though with 775 without success. 
It must be some safety feature from Docker I think

**Reply by trythatagain** [reply]
As I say, I'm no expert, but I'd like to ask a few questions, because this doesn't make sense. I expect it something simple though.

Is ssd an internal drive? Also can you create a directory or file manually on the ssd drive

Also what does `mount | grep "/dev/"` show.  On my system ,1 partition shows "/dev/nvme0n1p4 on /home type ext4 (rw,noatime,errors=remount-ro)" which shows it setup as 'rw'. It would be useful to double check that the drive is actually rw.

**Reply by louislam** [reply]
I believe it is a mount issue too. Actually I just googled and found some people were similar to you, you should check that out first.

`docker compose read-only file system`
